food-and-agriculture-organization-of-the-united-nations.csl

```diff
@@ -51,155 +63,287 @@
+                  <text variable="publisher-place" prefix=", "/>
+                </group>
+                <text macro="Cite-and-URL"/>
+              </else-if>
+              <else-if type="paper-conference">
+                <!-- conference paper -->
+                <text variable="title"/>
+                <group>
+                  <text term="in" text-case="capitalize-first" font-style="italic"/>
+                  <text macro="editor" prefix=" "/>
+                </group>
+                <text variable="container-title" font-style="italic"/>
+                <group suffix=".">
+                  <label suffix=" " variable="page" form="short"/>
+                  <text variable="page"/>
+                </group>
+                <text variable="collection-title"/>
+                <group suffix=".">
+                  <text term="presented at"/>
+                  <text variable="event" prefix=" "/>
+                  <date form="text" variable="issued" prefix=", "/>
+                  <text variable="publisher-place" prefix=", "/>
+                </group>
+                <text macro="Available-at"/>
+              </else-if>
+              <else>
+                <!-- other item types -->
+                <text variable="title" suffix="."/>
+                <text macro="publisher-and-place" suffix="."/>
+                <text macro="Cite-and-URL"/>
+              </else>
             </choose>
           </group>
-        </else-if>
-        <else-if type="thesis" match="any">
-          <text variable="title" font-style="italic" suffix="."/>
-          <text macro="publisher"/>
-          <text variable="genre" prefix=" (" suffix=")"/>
-        </else-if>
-        <else-if type="paper-conference" match="any">
-          <text variable="title" font-style="italic" suffix=". "/>
-          <text value="Paper presented at"/>
-          <group delimiter=", " prefix=" " suffix=".">
-            <text variable="event"/>
-            <text variable="event-place"/>
-            <date form="text" date-parts="year-month-day" variable="issued"/>
-          </group>
-          <choose>
-            <if match="any" variable="URL">
-              <text macro="Available-at-and-accessed" prefix=" "/>
-            </if>
-          </choose>
-        </else-if>
+        </if>
         <else>
-          <text variable="title" prefix=" " suffix="."/>
-          <group prefix=" ">
-            <text variable="container-title" form="short" font-style="italic" suffix=", "/>
-            <text variable="volume" form="short"/>
-            <text variable="issue" form="short" prefix="(" suffix=")"/>
+          <!-- case -->
+          <group delimiter=" ">
+            <group>
+              <text variable="title"/>
+              <text variable="volume" prefix=", "/>
+            </group>
+            <text variable="container-title"/>
+            <text variable="page"/>
+            <text variable="authority" prefix="("/>
```

Comment:
I have another question. To achieve the output that legal case is (_casetitle_, date) and all others are (author, date), I firstly wrote in the citation section without changing the author-short macro:
```
<layout prefix="(" suffix=")" delimiter="; ">
  <choose>
    <if match="none" type="legal_case"> 
      <text macro="author-short" suffix=", "/>
      <text macro="date"/>
    </if>
    <else>
      <text variable="title" font-style="italic"/>
      <text macro="date"/>
    </else>
  </choose>
</layout>
```
This renders fine when legal case is cited alone. However, if an (author, date) item precedes a case, the punctuation separating them becomes comma instead of semicolon: (author, date, _casetitle_, date), not (author, date; _casetitle_, date). I checked with other types and changed the author-short macro, and it worked fine, thus the last commit. Why is this?